HashRegistrarSimplified.sol

```diff
@@ -92,23 +92,49 @@ contract Deed {
     function () payable {}
 }
 
+
+/**
+ * @title Previous Registrar
+ * @dev The old registrar interface, for an upgrade path.
+ */
+contract PreviousRegistrar {
+    mapping (bytes32 => entry) public entries;
+
+    struct entry {
+        uint status;
+        Deed deed;
+        uint registrationDate;
+        uint value;
+        uint highestBid;
+    }
+
+    // Only exists for testing purposes
+    function register (bytes32 _hash) {
+        entry e = entries[_hash];
+        e.status = 2;
+        e.registrationDate = now;
+    }
+}
+
 /**
  * @title Registrar
  * @dev The registrar handles the auction process for each subnode of the node it owns.
  */
 contract Registrar {
     AbstractENS public ens;
     bytes32 public rootNode;
+    PreviousRegistrar public previousRegistrar;
```

Comment:
Shouldn't this code be in a new contract (possibly one that extends this one)? It's not ideal to have a bunch of unused upgrade code in the original registrar that has nothing to upgrade from.